pandas/tests/io/test_parquet.py

```diff
@@ -154,10 +154,46 @@ def check_round_trip(df, engine=None, path=None,
         write_kwargs['engine'] = engine
         read_kwargs['engine'] = engine
 
+    fastparquet_make_block_dtype = (
+        # Use of deprecated `dtype` in `make_block` that's hit only for
+        # bool dtypes with no Nones.
+        engine == 'fastparquet' and
+        LooseVersion(fastparquet.__version__) == LooseVersion("0.1.4") and
+        any(pd.api.types.is_bool_dtype(df[col]) for col in df.columns)
+    )
+
+    if (engine == 'pyarrow' and
```

Comment:
woa! pls make fixture for this, do it in a style that the existing follows. this is very hard to read.